theano/gpuarray/opt.py

```diff
@@ -1884,6 +1886,16 @@ def typebuild(dtype, broadcastable, context_name=context_name):
 def local_gpu_maxandargmax(op, context_name, inputs, outputs):
     return GpuMaxAndArgmax(op.get_params(None))
 
+
+# solve
+@register_opt('fast_compile')
+@op_lifter([slinalg.Solve])
+@register_opt2([theano.tensor.slinalg.Solve], 'fast_compile')
+def local_gpu_solve(op, context_name, inputs, outputs):
+    if not cusolver_available:
+        return
+    return GpuCusolverSolve()
```

Comment:
small comment, building new object in Python is costly, so this would be faster if one `gpu_cusolver_solve = GpuCusolverSolve()` is created in the file that create this op and reused here. So this will reuse always the same instance.

This do not block this PR.